Fix for MenuFlyoutItem stops working after navigating away from and back to page

BagavathiPerumal opened the pull request:
### Root cause

The issue occurs because the BindingContext is set to null after navigating to another page. As a result, when returning to the page, the MenuFlyoutItem loses its binding, causing the Command to stop functioning properly. 

### Description of Issue Fix

The fix involves the parent of the MenuBarItem only set to null when the parent of MenuBarItem is MenuBar instead of page. This prevents the unintended loss of the BindingContext, which can disrupt bindings and commands when navigating back to a page.

Tested the behavior in the following platforms.
 
- [x] Windows
- [x] Mac
- [ ] iOS
- [ ] Android

The MenuBar control is not supported on iOS and Android, so this fix was not tested on these platforms.

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes https://github.com/dotnet/maui/issues/21675

<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

### Output

| Before Issue Fix | After Issue Fix |
|----------|----------|
| <video width="270" height="600"  src="https://github.com/user-attachments/assets/ccc01455-085d-4dff-b5b2-41ea273534d9"> | <video width="270" height="600"  src="https://github.com/user-attachments/assets/91ae9baf-7097-492d-a638-139e566e1b4b"> |

BagavathiPerumal committed:
fix-21675-Updated the BindingContext of MenuBarItem after page navigation.

BagavathiPerumal committed:
fix-21675-Updated the BindingContext of MenuBarItem after page navigation.

BagavathiPerumal committed:
fix-21675- Added UI testcase.

BagavathiPerumal committed:
fix-21675- Added UI testcase.

BagavathiPerumal committed:
fix-21675-Removed unwanted space.

BagavathiPerumal committed:
fix-21675-Removed unwanted space.

BagavathiPerumal committed:
Merge branch 'main' of https://github.com/BagavathiPerumal/maui into fix-21675

BagavathiPerumal committed:
fix-21675-removed unwanted spacing.

BagavathiPerumal committed:
fix-21675-removed unwanted spacing.

BagavathiPerumal committed:
fix-21675-Updated testcase.

BagavathiPerumal committed:
fix-21675-Updated testcase.

BagavathiPerumal committed:
fix-21675-Updated testcase script.

jsuarezruiz commented:
/azp run

BagavathiPerumal commented:
/rebase

jsuarezruiz commented:
/azp run

BagavathiPerumal commented:
/rebase

PureWeen reviewed (changes requested):
It doesn't look like this needs a UITest

Can you move this to a unit test? 
https://github.com/dotnet/maui/blob/90d039638a8b58a1d5df3c5aa0babac49b1fd002/src/Controls/tests/Core.UnitTests/MenuItemTests.cs#L10

BagavathiPerumal commented:
> It doesn't look like this needs a UITest
> 
> Can you move this to a unit test?
> 
> https://github.com/dotnet/maui/blob/90d039638a8b58a1d5df3c5aa0babac49b1fd002/src/Controls/tests/Core.UnitTests/MenuItemTests.cs#L10

@PureWeen, we couldn't recreate the same UI test scenario in the unit test case because the command of the MenuFlyoutItem is always present after page navigation.
 
[Fact]
public async void UpdateMenuBarItemBindingContext()
{
	bool fired = false;
	var cmd = new Command(() => fired = true);
	var bindingContext = new
	{
		MenuItemCommand = cmd
	};
 
	var menuFlyoutItem = new MenuFlyoutItem();
	var menuItem = new MenuBarItem { menuFlyoutItem };
	var menubarItems = new List<MenuBarItem> { menuItem };
	var mainPage = new ContentPage
	{
		MenuBarItems = menubarItems,
	};
	mainPage.BindingContext = bindingContext;
	menuFlyoutItem.SetBinding(MenuFlyoutItem.CommandProperty, new Binding("MenuItemCommand"));
 
	var page = new ContentPage
	{
		BindingContext = bindingContext
	};
 
	NavigationPage nav = new NavigationPage(mainPage);
	await mainPage.Navigation.PushAsync(page);
	await page.Navigation.PopAsync();
 
	menuFlyoutItem.Command.Execute(null);
 
	Assert.True(fired);
}

BagavathiPerumal committed:
fix-21675-Updated testcase which only execute for the Windows and MacOS

BagavathiPerumal committed:
Merge branch 'main' of https://github.com/BagavathiPerumal/maui into fix-21675

BagavathiPerumal committed:
fix-21675-Conflict resolved.

jsuarezruiz commented:
/azp run

BagavathiPerumal committed:
Merge branch 'main' of https://github.com/BagavathiPerumal/maui into fix-21675

jsuarezruiz commented:
/azp run

BagavathiPerumal committed:
Merge branch 'main' of https://github.com/BagavathiPerumal/maui into fix-21675

jsuarezruiz commented:
/azp run

jsuarezruiz commented on src/Controls/src/Core/Page/Page.cs:
```diff
@@ -449,6 +449,26 @@ protected virtual void LayoutChildren(double x, double y, double width, double h
+			{
+				if (menuBarItem.BindingContext is null)
+				{
+					SetInheritedBindingContext(menuBarItem, BindingContext);
```
Isn't the BindingContext already propagated to MenuBarItems here?
https://github.com/dotnet/maui/blob/d541b54292554f270919e1c491fae7618a58ea62/src/Controls/src/Core/Page/Page.cs#L496

BagavathiPerumal replied:
@jsuarezruiz, the BindingContext of MenuBarItem is set to null after navigating to another page. As a result, when returning to the page, the MenuFlyoutItem loses its binding, causing the Command to stop functioning properly. Therefore, the BindingContext of MenuBarItem is updated within the OnAppearing() method of the page.

rmarinho replied:
How is it set the first time when you navigate to the page?

mattleibow replied:
Do we know why it is set to null? The menu items are still part of the page, and the page still has the same binding context. Why is navigating away changing the binding context?

BagavathiPerumal replied:
@mattleibow, The MenuBarItem's parent is set to null in the RemoveAt method of the MenuBar class during the item removal process. 
https://github.com/dotnet/maui/blob/2478901a5b74b4298f7397e8554d897bdd04effe/src/Controls/src/Core/Menu/MenuBar.cs#L117
 
Since the BindingContext is inherited from the parent, this action causes the BindingContext to be lost. Although the menu items remain part of the page, navigating away triggers this removal logic, detaching the MenuBarItem from its parent and resetting its BindingContext. This behavior occurs because the MenuBar explicitly clears the parent during navigation, which unintentionally impacts the binding.

PureWeen replied:
I think that this is where the actual bug is @BagavathiPerumal 

The fix here might be to check if the `Parent` is `MenuBar` and if it's not then don't set it to null. 

The Parent of `MenuBarItem` is the Page that it's on so the `Page` needs to control when the parent is set to null

BagavathiPerumal replied:
@PureWeen, As suggested, setting the parent of a MenuBarItem to null when its parent is MenuBar caused the following test case to fail.
 
[ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146)
 
```
public void RemoveAt(int index)
{
   var item = _menus[index];
 
   if (index < 0)
       return;
       
   _menus.RemoveAt(index);
   NotifyHandler(nameof(IMenuBarHandler.Remove), index, item);
   
   if (item is Element e && e.Parent is MenuBar)
         e.Parent = null;
}
```
 
Additionally, based on the comment in the test case linked below, it seems that MenuBarItem objects are currently parented only to Page instances instead of the MenuBar. Could you provide suggestions or guidance regarding this behavior?
 
https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L112

PureWeen commented on src/Controls/tests/TestCases.HostApp/Issues/Issue21675.cs:
```diff
@@ -0,0 +1,162 @@
+		}
+	}
+
+	public class MainPage : ContentPage
```
Can you move this class to be an internal class of `Issue21675` so it's not part of the `Maui.Controls.Sample.Issues` NS

BagavathiPerumal replied:
@PureWeen, I have moved these classes to Issue21675.

BagavathiPerumal committed:
fix-21675-Moved testcase classes to Issue21675.

jsuarezruiz commented:
/rebase

jsuarezruiz commented:
/azp run

rmarinho commented on src/Controls/src/Core/Page/Page.cs:
```diff
@@ -449,6 +449,26 @@ protected virtual void LayoutChildren(double x, double y, double width, double h
+		/// <summary>
+		/// Resets the BindingContext for each MenuBarItem when it becomes null after page navigation.
+		/// </summary>
+		private void UpdateMenuBarItemBindingContext()
```
```suggestion
	 void UpdateMenuBarItemBindingContext()
```

please don t use private

rmarinho committed:
Update src/Controls/src/Core/Page/Page.cs

rmarinho commented:
/azp run

rmarinho commented:
/rebase

BagavathiPerumal committed:
Merge branch 'dotnet:main' into fix-21675

BagavathiPerumal commented:
/rebase

BagavathiPerumal commented:
/rebase

BagavathiPerumal committed:
Merge branch 'main' of https://github.com/BagavathiPerumal/maui into fix-21675

PureWeen reviewed (changes requested):
Can we validate this one using a unit test vs a UI Test?

BagavathiPerumal commented:
> Can we validate this one using a unit test vs a UI Test?

@PureWeen, As discussed here in this [comment](https://github.com/dotnet/maui/pull/25170#issuecomment-2416771495), we were facing difficulties in converting the test from UI to Unit testcase.

PureWeen commented:
> > Can we validate this one using a unit test vs a UI Test?
> 
> @PureWeen, As discussed here in this [comment](https://github.com/dotnet/maui/pull/25170#issuecomment-2416771495), we were facing difficulties in converting the test from UI to Unit testcase.

If you wire up through a `TestWindow` does it work? 

For example, if you check any of tests under `NavigationModelTests`

BagavathiPerumal commented:
> > > Can we validate this one using a unit test vs a UI Test?
> > 
> > 
> > @PureWeen, As discussed here in this [comment](https://github.com/dotnet/maui/pull/25170#issuecomment-2416771495), we were facing difficulties in converting the test from UI to Unit testcase.
> 
> If you wire up through a `TestWindow` does it work?
> 
> For example, if you check any of tests under `NavigationModelTests`

@PureWeen, As suggested, when wiring up through a TestWindow in the UnitTest case, we couldn't recreate the UI test scenario because the MenuFlyoutItem command remains active after page navigation. Could you please share your suggestions.
 
 
```
[Fact]
public async void UpdateMenuBarItemBindingContext()
{
	bool fired = false;
	var cmd = new Command(() => fired = true);
	var bindingContext = new
	{
		MenuItemCommand = cmd
	};
 
	MenuFlyoutItem flyout;
	MenuBarItem menuItem;
	var mainPage = new ContentPage
	{
		MenuBarItems =
		{
			(menuItem = new MenuBarItem
			{
				(flyout = new MenuFlyoutItem { })
			})
		}
	};
	mainPage.BindingContext = bindingContext;
	flyout.SetBinding(MenuFlyoutItem.CommandProperty, new Binding("MenuItemCommand"));
 
	var page = new ContentPage
	{
		BindingContext = bindingContext
	};
 
	TestWindow testWindow = new TestWindow(mainPage);
	NavigationPage nav = new NavigationPage(mainPage);
	await mainPage.Navigation.PushAsync(page);
	await page.Navigation.PopAsync();
 
	flyout.Command.Execute(null);
 
	Assert.True(fired);
}
```

PureWeen commented:
> > > > Can we validate this one using a unit test vs a UI Test?
> > > 
> > > 
> > > @PureWeen, As discussed here in this [comment](https://github.com/dotnet/maui/pull/25170#issuecomment-2416771495), we were facing difficulties in converting the test from UI to Unit testcase.
> > 
> > 
> > If you wire up through a `TestWindow` does it work?
> > For example, if you check any of tests under `NavigationModelTests`
> 
> @PureWeen, As suggested, when wiring up through a TestWindow in the UnitTest case, we couldn't recreate the UI test scenario because the MenuFlyoutItem command remains active after page navigation. Could you please share your suggestions.
> 
> ```
> [Fact]
> public async void UpdateMenuBarItemBindingContext()
> {
> 	bool fired = false;
> 	var cmd = new Command(() => fired = true);
> 	var bindingContext = new
> 	{
> 		MenuItemCommand = cmd
> 	};
>  
> 	MenuFlyoutItem flyout;
> 	MenuBarItem menuItem;
> 	var mainPage = new ContentPage
> 	{
> 		MenuBarItems =
> 		{
> 			(menuItem = new MenuBarItem
> 			{
> 				(flyout = new MenuFlyoutItem { })
> 			})
> 		}
> 	};
> 	mainPage.BindingContext = bindingContext;
> 	flyout.SetBinding(MenuFlyoutItem.CommandProperty, new Binding("MenuItemCommand"));
>  
> 	var page = new ContentPage
> 	{
> 		BindingContext = bindingContext
> 	};
>  
> 	TestWindow testWindow = new TestWindow(mainPage);
> 	NavigationPage nav = new NavigationPage(mainPage);
> 	await mainPage.Navigation.PushAsync(page);
> 	await page.Navigation.PopAsync();
>  
> 	flyout.Command.Execute(null);
>  
> 	Assert.True(fired);
> }
> ```

@BagavathiPerumal  in your test you aren't assigning the right thing to the window

You need to change those lines to this
```C#
NavigationPage nav = new NavigationPage(mainPage);
			TestWindow testWindow = new TestWindow(nav);
			await mainPage.Navigation.PushAsync(page);
			await page.Navigation.PopAsync();
		
```

PureWeen commented:
/rebase

BagavathiPerumal committed:
fix-21675-Added UnitTestcase and removed UI testcase.

BagavathiPerumal commented:
> > > > > Can we validate this one using a unit test vs a UI Test?
> > > > 
> > > > 
> > > > @PureWeen, As discussed here in this [comment](https://github.com/dotnet/maui/pull/25170#issuecomment-2416771495), we were facing difficulties in converting the test from UI to Unit testcase.
> > > 
> > > 
> > > If you wire up through a `TestWindow` does it work?
> > > For example, if you check any of tests under `NavigationModelTests`
> > 
> > 
> > @PureWeen, As suggested, when wiring up through a TestWindow in the UnitTest case, we couldn't recreate the UI test scenario because the MenuFlyoutItem command remains active after page navigation. Could you please share your suggestions.
> > ```
> > [Fact]
> > public async void UpdateMenuBarItemBindingContext()
> > {
> > 	bool fired = false;
> > 	var cmd = new Command(() => fired = true);
> > 	var bindingContext = new
> > 	{
> > 		MenuItemCommand = cmd
> > 	};
> >  
> > 	MenuFlyoutItem flyout;
> > 	MenuBarItem menuItem;
> > 	var mainPage = new ContentPage
> > 	{
> > 		MenuBarItems =
> > 		{
> > 			(menuItem = new MenuBarItem
> > 			{
> > 				(flyout = new MenuFlyoutItem { })
> > 			})
> > 		}
> > 	};
> > 	mainPage.BindingContext = bindingContext;
> > 	flyout.SetBinding(MenuFlyoutItem.CommandProperty, new Binding("MenuItemCommand"));
> >  
> > 	var page = new ContentPage
> > 	{
> > 		BindingContext = bindingContext
> > 	};
> >  
> > 	TestWindow testWindow = new TestWindow(mainPage);
> > 	NavigationPage nav = new NavigationPage(mainPage);
> > 	await mainPage.Navigation.PushAsync(page);
> > 	await page.Navigation.PopAsync();
> >  
> > 	flyout.Command.Execute(null);
> >  
> > 	Assert.True(fired);
> > }
> > ```
> 
> @BagavathiPerumal in your test you aren't assigning the right thing to the window
> 
> You need to change those lines to this
> 
> ```cs
> NavigationPage nav = new NavigationPage(mainPage);
> 			TestWindow testWindow = new TestWindow(nav);
> 			await mainPage.Navigation.PushAsync(page);
> 			await page.Navigation.PopAsync();
> 		
> ```

@PureWeen, Thank you for the suggestion. I have added the unit test case accordingly. Please review and let me know if you have any further concerns.

PureWeen reviewed (changes requested):
My comment here still applies to the PR in its current state

https://github.com/dotnet/maui/pull/25170#discussion_r1915619320

PureWeen commented:
/rebase

PureWeen commented:
@BagavathiPerumal 
Apologies but I'm having trouble following the thread on this one

AFAICT this code is wrong
https://github.com/dotnet/maui/blob/2478901a5b74b4298f7397e8554d897bdd04effe/src/Controls/src/Core/Menu/MenuBar.cs#L116-L117

and should be changed to this

```C#
			if (item is Element e && e.Parent == this)
				e.Parent = null;
```

What I'm thinking here is

1) remove the code you've added
2) make the change I link above

And then that should be the proper fix here. 

If I'm still missing something here can you summarize it all in a comment so it's a bit easier to follow where I'm not understanding?

BagavathiPerumal commented:
> @BagavathiPerumal Apologies but I'm having trouble following the thread on this one
> 
> AFAICT this code is wrong
> 
> https://github.com/dotnet/maui/blob/2478901a5b74b4298f7397e8554d897bdd04effe/src/Controls/src/Core/Menu/MenuBar.cs#L116-L117
> 
> and should be changed to this
> 
> ```cs
> 			if (item is Element e && e.Parent == this)
> 				e.Parent = null;
> ```
> 
> What I'm thinking here is
> 
> 1. remove the code you've added
> 2. make the change I link above
> 
> And then that should be the proper fix here.
> 
> If I'm still missing something here can you summarize it all in a comment so it's a bit easier to follow where I'm not understanding?

@PureWeen, As suggested, setting the parent of a MenuBarItem to null when its parent is MenuBar resolves the actual [issue](https://github.com/dotnet/maui/issues/21675). However, it causes the following test case to fail.
 
[ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146)
 
In this test case, the ContentPage's MenuBarItems are cleared. At that time, the parent of the MenuBarItem is ContentPage. Therefore, when MenuBarItems are cleared, the parent of the MenuBarItem must also be cleared and set to null. This seems to be a valid case. However, due to your suggested fix, these test cases have failed. 
 
The actual [issue](https://github.com/dotnet/maui/issues/21675) is that the MenuBarItem's BindingContext is reset to null after navigating to another page due to the previous page's MenuBarItem parent being set to null. To resolve this, I added a fix where the BindingContext for the MenuBarItem is reapplied within the OnAppearing() method of the page. This ensures that the MenuBarItem's BindingContext is correctly set when the page is displayed again after navigation, without affecting any previous logic.

PureWeen commented:
> > @BagavathiPerumal Apologies but I'm having trouble following the thread on this one
> > AFAICT this code is wrong
> > https://github.com/dotnet/maui/blob/2478901a5b74b4298f7397e8554d897bdd04effe/src/Controls/src/Core/Menu/MenuBar.cs#L116-L117
> > 
> > and should be changed to this
> > ```cs
> > 			if (item is Element e && e.Parent == this)
> > 				e.Parent = null;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > What I'm thinking here is
> > 
> > 1. remove the code you've added
> > 2. make the change I link above
> > 
> > And then that should be the proper fix here.
> > If I'm still missing something here can you summarize it all in a comment so it's a bit easier to follow where I'm not understanding?
> 
> @PureWeen, As suggested, setting the parent of a MenuBarItem to null when its parent is MenuBar resolves the actual [issue](https://github.com/dotnet/maui/issues/21675). However, it causes the following test case to fail.
> 
> [ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146)
> 
> In this test case, the ContentPage's MenuBarItems are cleared. At that time, the parent of the MenuBarItem is ContentPage. Therefore, when MenuBarItems are cleared, the parent of the MenuBarItem must also be cleared and set to null. This seems to be a valid case. However, due to your suggested fix, these test cases have failed.
> 
> The actual [issue](https://github.com/dotnet/maui/issues/21675) is that the MenuBarItem's BindingContext is reset to null after navigating to another page due to the previous page's MenuBarItem parent being set to null. To resolve this, I added a fix where the BindingContext for the MenuBarItem is reapplied within the OnAppearing() method of the page. This ensures that the MenuBarItem's BindingContext is correctly set when the page is displayed again after navigation, without affecting any previous logic.

Yea, but the problem is that the `MenuBarItems` parent should never be set to null. It belongs to the the `Page` so it isn't the responsibility of the `MenuBar` to clear out the parent. 

The tests inside  [ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146) attempt to be generalized across all the different types of menu since they apply to a lot of the same scenarios but there are cases where the logic differs. 

For example, if you check this test
https://github.com/dotnet/maui/blob/a5553ad0f101b93d00ce67a6c2a25ccecb47dad4/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L94-L100

you'll see the logic deviates based on the type of item, so I'm pretty sure this is the same case. The `ClearUpdatesParent` needs to be fixed so that it's actually checking that the parent isn't removed when clearing the `MenuBar`

BagavathiPerumal committed:
fix-21675- Parent of the MenuBarItems only null when the parent is MenuBar instead of page.

BagavathiPerumal commented:
> > > @BagavathiPerumal Apologies but I'm having trouble following the thread on this one
> > > AFAICT this code is wrong
> > > https://github.com/dotnet/maui/blob/2478901a5b74b4298f7397e8554d897bdd04effe/src/Controls/src/Core/Menu/MenuBar.cs#L116-L117
> > > 
> > > and should be changed to this
> > > ```cs
> > > 			if (item is Element e && e.Parent == this)
> > > 				e.Parent = null;
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > What I'm thinking here is
> > > 
> > > 1. remove the code you've added
> > > 2. make the change I link above
> > > 
> > > And then that should be the proper fix here.
> > > If I'm still missing something here can you summarize it all in a comment so it's a bit easier to follow where I'm not understanding?
> > 
> > 
> > @PureWeen, As suggested, setting the parent of a MenuBarItem to null when its parent is MenuBar resolves the actual [issue](https://github.com/dotnet/maui/issues/21675). However, it causes the following test case to fail.
> > [ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146)
> > In this test case, the ContentPage's MenuBarItems are cleared. At that time, the parent of the MenuBarItem is ContentPage. Therefore, when MenuBarItems are cleared, the parent of the MenuBarItem must also be cleared and set to null. This seems to be a valid case. However, due to your suggested fix, these test cases have failed.
> > The actual [issue](https://github.com/dotnet/maui/issues/21675) is that the MenuBarItem's BindingContext is reset to null after navigating to another page due to the previous page's MenuBarItem parent being set to null. To resolve this, I added a fix where the BindingContext for the MenuBarItem is reapplied within the OnAppearing() method of the page. This ensures that the MenuBarItem's BindingContext is correctly set when the page is displayed again after navigation, without affecting any previous logic.
> 
> Yea, but the problem is that the `MenuBarItems` parent should never be set to null. It belongs to the the `Page` so it isn't the responsibility of the `MenuBar` to clear out the parent.
> 
> The tests inside [ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146) attempt to be generalized across all the different types of menu since they apply to a lot of the same scenarios but there are cases where the logic differs.
> 
> For example, if you check this test
> 
> https://github.com/dotnet/maui/blob/a5553ad0f101b93d00ce67a6c2a25ccecb47dad4/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L94-L100
> 
> you'll see the logic deviates based on the type of item, so I'm pretty sure this is the same case. The `ClearUpdatesParent` needs to be fixed so that it's actually checking that the parent isn't removed when clearing the `MenuBar`

@PureWeen, As suggested, I have modified the fix and the test case([ClearUpdatesParent()](https://github.com/dotnet/maui/blob/1a4f263f8951ebb591b2bc0ccb3ebba340c0a9cb/src/Controls/tests/Core.UnitTests/Menu/MenuTestBase.cs#L146)). Could you please share your concerns about this.

jsuarezruiz commented:
/azp run

HimanshuBansal112 commented:
Does it fix https://github.com/dotnet/maui/issues/13848, repository for my issue is at https://github.com/HimanshuBansal112/NetMauiIssue for testing?
